src/components/Filters.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useState, ChangeEvent } from 'react'
+import MUISelect from '@material-ui/core/Select'
+import MenuItem from '@material-ui/core/MenuItem'
+import Box from '@material-ui/core/Box'
+import FormControl from '@material-ui/core/FormControl'
+import InputLabel from '@material-ui/core/InputLabel'
+import uniq from 'lodash/uniq'
+import { withStyles } from '@material-ui/core/styles'
+import { capitalize } from '../helpers'
+import { IProps } from '../containers/FilterContainer'
+import { INamespaceAPI } from '../types/api'
+
+const StyledFormControl = withStyles({
+  root: {
+    margin: '0rem 2rem',
+    minWidth: '10rem'
+  }
+})(FormControl)
+
+type IEntity = INamespaceAPI | string
+interface IFilterDictionary {
+  [key: string]: {
+    entities: IEntity[]
+    accessor: (n: IEntity) => string
+    default: any
+  }
+}
+
+const filterByOptions: { [key: string]: 'namespace' | 'sourceName' } = {
+  /* display name: entitiy name on dataset schema */
+  namespace: 'namespace',
+  datasource: 'sourceName'
+}
+
+const Filters = (props: IProps): ReactElement => {
+  const { namespaces, datasets, filterJobs, filterDatasets, showJobs } = props
+
+  const [currentFilter, setCurrentFilter] = useState('all')
+  const [currentFilterValue, setCurrentFilterValue] = useState({})
+  const [showSubFilter, toggleShowSubFilter] = useState(false)
+
+  const datasources = uniq(datasets.map(d => d.sourceName))
+  const filterDictionary: IFilterDictionary = {
+    namespace: {
+      entities: namespaces,
+      accessor: n => (n as INamespaceAPI).name,
+      default: { name: '' }
+    },
+    datasource: {
+      entities: datasources,
+      accessor: (n: string): string => n,
+      default: ''
+    }
+    /* Can add more filters here */
+  }
+
+  /* Set the category we will be filtering by */
+  const onPrimaryFilterChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const newPrimaryFilter = e.target.value as 'namespace' | 'datasource' | 'all'
+    console.log('NEW PRI<AR', newPrimaryFilter)
+    setCurrentFilter(newPrimaryFilter)
+    if (newPrimaryFilter === 'all') {
+      toggleShowSubFilter(false)
```

Comment:
maybe rename this for clarity?  I would assume a toggle function wouldn't take an argument?

Reply:
Totally agree with this!  Will change.